### PR TITLE
Fix world-size-one aliasing in MLP batch sync

### PR DESCRIPTION
**Pull request by merrymercy**
## Motivation

At world size one, `expand().contiguous()` can return storage that still aliases the fallback tensor, so later masked writes can corrupt the fallback values they read.

## Modifications

Allocate the gathered fallback buffer with `repeat()` so it always owns independent storage, including the world-size-one case.

## Accuracy Tests

- `uv run python3 -m py_compile python/sglang/srt/managers/scheduler_components/dp_attn.py`
- `uv run python3 -c "import torch; fallback = torch.arange(7); gathered = fallback.repeat(1, 1, 1); gathered.zero_(); assert fallback.tolist() == list(range(7))"`

## Speed Tests and Profiling

Not run; the change only affects construction of a small synchronization metadata buffer.

## Checklist

- [x] `with-proxy uv run pre-commit run --files python/sglang/srt/managers/scheduler_components/dp_attn.py`
- [x] No documentation update is required for this bug fix.

## Original commits

- `778b30ce0`

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #31976780289](https://github.com/sgl-project/sglang/actions/runs/31976780289)
Latest PR Test (Extra): :x: [Run #31976780111](https://github.com/sgl-project/sglang/actions/runs/31976780111)

### CODE REVIEW

**Review by merrymercy — COMMENTED** [review]
approve
